packages/babel-traverse/src/path/context.ts

```diff
@@ -133,6 +133,29 @@ export function stop(this: NodePath) {
   this._traverseFlags |= SHOULD_SKIP | SHOULD_STOP;
 }
 
+export function _forceSetScope(current: NodePath) {
+  let path = current.parentPath;
+
+  if (
+    // Skip method scope if is computed method key or decorator expression
+    ((current.key === "key" || current.listKey === "decorators") &&
+      path.isMethod()) ||
```

Comment:
Could you add tests covering this change? It seems to me that tests are passing without this change.